### PR TITLE
[TS] LPS-154862 Unable to manage a site without the user locale enabled

### PR DESCRIPTION
**Pull request by joseluisbango**
Relevant issues:

https://issues.liferay.com/browse/LPS-154862
https://issues.liferay.com/browse/PTR-3085

---

Hi @liferay-usm,

Could you review this pull?

Previously, I sent it to core infra team (see https://github.com/liferay-core-infra/liferay-portal/pull/961) but they think you should review it instead.

As we discussed in PTR-3085 with Lima and Echo teams, users should be able to manage a Site using any language as long as it is available in the instance. Currently, they can only do that if they go to Site Admin from Control Panel > Sites. If they navigate to Site Admin from a different place, such us the Site Selector shown when clicking the compass icon (what they usually do), they cannot change the display language.

They were able to do this in 6.2 manually adding the language code to the URL, so they perceive that as loss of functionality.

I have added a property as requested in the PTR so that the users can see the Site Admin pages (`/~/control_panel` pages) in any instance-available language. In addition, I have changed the `user_locale_options.jsp` to allow the users to easily change to their language from the alert notification.

Let me know if you have any questions.

Thanks!

### CODE REVIEW

**Comment by Sharryxue** [comment]
No related failure found in ci:test:relevant test suite.

**Review by drewbrokke — APPROVED** [review]
This change seems good to me. Are you able to add test coverage for this scenario as well? If so, please send it in a follow-up PR.

**Comment by drewbrokke** [comment]
@joseluisbango FYI looks like there's a couple of merge conflicts.

**Comment by joseluisbango** [comment]
Hi @drewbrokke @pei-jung ,

Thanks for the review! I have rebased and force-pushed the changes. Could it be moved forward then?

Regards

**Comment by drewbrokke** [comment]
ci:test:sf

File: portal-impl/src/com/liferay/portal/util/PortalImpl.java [hunk 1]
```diff
@@ -6048,6 +6048,17 @@ public boolean isCompanyControlPanelPortlet(
...
+	@Override
+	public boolean isControlPanelPath(String path) {
+		if (Validator.isNull(path)) {
+			return false;
+		}
+
+		return path.contains(
+			VirtualLayoutConstants.CANONICAL_URL_SEPARATOR +
+				GroupConstants.CONTROL_PANEL_FRIENDLY_URL);
+	}
```
Comment (drewbrokke) [inline portal-impl/src/com/liferay/portal/util/PortalImpl.java:6060]: I think the naming of this method is not quite right for what it does. The paths to company level control panel portlets such as `Users and Organizations` do not contain the canonical URL separator, but those paths should still be considered control panel paths. 

I think we should do one of two things:
1. Change the method to something like `isGroupControlPanelPath`. That way it's clear that we are only dealing with group paths.
2. Change the logic to not require the canonical separator. Then all control panel paths will validate as expected.

**Review by drewbrokke — CHANGES_REQUESTED** [review]
@joseluisbango I have one small suggestion, please see my inline comment.

File: portal-impl/src/com/liferay/portal/util/PortalImpl.java (hunk 1 above)
Comment (joseluisbango) [reply portal-impl/src/com/liferay/portal/util/PortalImpl.java:6060]: Yes, I agree with you. I am going to rename the method to `isGroupControlPanelPath` so it is clearer and we can keep the current logic.

**Comment by joseluisbango** [comment]
Hi @drewbrokke, thanks for your suggestion, totally agree. I have force-pushed the requested changes.

**Comment by joseluisbango** [comment]
ci:forward

**Comment by drewbrokke** [comment]
@epark18 are these failures related? If not I think we can manually forward this one.

**Comment by epark18** [comment]
Failures do not appear to be related.
Forwarded to Brian. https://github.com/brianchandotcom/liferay-portal/pull/120369

**Comment by epark18** [comment]
ci:close

**Pull request closed without merging**